lib/Driver/Driver.cpp

```diff
@@ -2217,6 +2217,16 @@ Job *Driver::buildJobsForAction(Compilation &C, const JobAction *JA,
     PrimaryInput = Out.getPrimaryOutputFilenames()[i];
   }
 
+  // With -index-file option, the primary input is the one passed with
+  // -index-file-path.
```

Comment:
Would it be possible to add a pointer here to a larger comment or a document that describes how -index-file-path works? How it is used, or how such an invocation differs from normal? I don't really understand this stuff and don't know where to look and the next person who has to work on the driver may have the same problem. Thanks.

Reply:
+1 for more documentation of how this mode is supposed to work.  I admit I don't fully understand it either.  I will talk to @akyrtzi  about addressing this as a follow-up commit.